### ISSUE TITLE
Use sovereign accounts of sending chains to pay relayer rewards

### ISSUE THREAD

**Issue by svyatonik** [paritytech/parity-bridges-common#2585]
In `bridges-v1` we are using `into_account_truncating` on [`RewardsAccountParams`](https://github.com/paritytech/parity-bridges-common/blob/master/primitives/relayers/src/lib.rs#L54-L71) structure to compute `AccountId` that will be used to pay relayer rewards. 

This is not XCM-friendly. Also it means that the sending chain (e.g. asset hub) needs to maintain two accounts at bridge hub - one for paying XCM fees and another to pay relayer rewards. Let's remove `RewardsAccountParams` and use SA of sending chain for both purposes.

**Comment by bkontur** [comment]
@svyatonik 
Can I take it or are you working  on that?
I've just added claim rewards stuff to Ro/Wo demo scripts :) https://github.com/paritytech/polkadot-sdk/pull/1215/commits/559b954fe88941498e3cecfeab785d08f6c36cff

**Comment by svyatonik** [comment]
@bkontur No, I'm not working on that now - please take it if you want :)

**Comment by bkontur** [comment]
xcm fees will be also paid from SA of asset hub on BridgeHub,
so it is ok to use the same SA account for claiming rewards?

**Comment by svyatonik** [comment]
I think - it is fine. That's the point of this issue. But if you have some concerns - please share. I may be wrong here

**Comment by bkontur** [comment]
no concerns, it simplify stuff, and it also make sense that SA of AssetHub manages rewards.
just wonder if it is ok from security-audit point of view.

**Comment by svyatonik** [comment]
> just wonder if it is ok from security-audit point of view.

I'm not the right person to answer that :) We may impl-then-ask or ask-then-impl. But from my PoV, I see no problems with that.

**Comment by acatangiu** [comment]
@bkontur let's prioritize this post initial launch and get it in a subsequent patch release
